### PR TITLE
react design

### PR DESCRIPTION
**Pull request by AaronHe09**
image

### CODE REVIEW

**Review by TimelikeClosure — CHANGES_REQUESTED** [review]
@AaronHe09 Phase 1 looks good, looking forward to Phase 2

File: react-design/my-app/src/IndexButtons.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+export default function IndexButtons({ items }) {
+  return (
+    <div>
+      {items.map((item) => (
+        <button key={item}>{items.indexOf(item)}</button>
+      ))}
+    </div>
+  );
+}
```
Comment (TimelikeClosure) [inline react-design/my-app/src/IndexButtons.js:9]: Don't forget that the currently-selected item's button should be highlighted:

---

image

**Review by TimelikeClosure — CHANGES_REQUESTED** [review]

File: react-design/my-app/src/IndexButtons.js (hunk 1 above)
Comment (AaronHe09) [reply react-design/my-app/src/IndexButtons.js:9]: are you not suppose to use an event handler function for that?

File: react-design/my-app/src/RotatingBanner.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import Banner from './Banner';
+import { useState } from 'react';
+
+export default function RotatingBanner({ items }) {
+  const [currentIndex, setCurrentIndex] = useState(0);
+
+  function handleButtonsOnClick(e) {
+    const int = parseInt(e.target.textContent);
+    setCurrentIndex(int);
+  }
```
Comment (TimelikeClosure) [inline react-design/my-app/src/RotatingBanner.js:12]: I generally recommend against passing full event objects up the component tree. Instead, only pass the value the parent component needs (in this case, the index).

**Review by TimelikeClosure — CHANGES_REQUESTED** [review]

**Review by TimelikeClosure — APPROVED** [review]

**Pull request merged**